### ISSUE TITLE
Dhall.Pretty updated; fails to build on Nix

### ISSUE THREAD

**Issue by toastal** [purescript/spago#738]
```
src/Spago/Dhall.hs:39:11: error:
    • Couldn't match expected type ‘Maybe Dhall.Pretty.CharacterSet’
                  with actual type ‘Dhall.Pretty.CharacterSet’
    • In the first argument of ‘Dhall.Format.Format’, namely
        ‘Dhall.Pretty.ASCII’
      In the second argument of ‘(.)’, namely
        ‘Dhall.Format.Format
           Dhall.Pretty.ASCII NoCensor
           (PossiblyTransitiveInputFile (Text.unpack pathText) NonTransitive)’
      In the expression:
        Dhall.Format.format
          . Dhall.Format.Format
              Dhall.Pretty.ASCII NoCensor
              (PossiblyTransitiveInputFile (Text.unpack pathText) NonTransitive)
   |
39 |           Dhall.Pretty.ASCII
   |           ^^^^^^^^^^^^^^^^^^
```

I believe this came up in the discussion in ASCII vs. Unicode [discussion](https://discourse.dhall-lang.org/t/change-dhall-format-to-use-ascii-by-default/115/24) on the Dhall Discourse, but now the autoformatter can deduce which style the user is using. The update happened in `dhall-haskell` last month with [this commit](https://github.com/dhall-lang/dhall-haskell/commit/fa27901998e51399254f1c907126b3cd56da5c0c). I suppose this library at least needs the library updated.

I would assume `spago` should also be "smart" about formatting too even if PureScript, the org?, wants to keep ASCII by default?

**Comment by f-f** [comment]
I don't have a strong preference for this, so using the new encoding detection machinery sounds good to me.
I have no plans to update to the new Dhall version myself soon, but I'll gladly accept a PR for that change 🙂

**Comment by toastal** [comment]
I wish I was a Haskell programmer so I could help. I may have to switch `spago` to `easy-purescript-nix` in the meantime.

**Comment by f-f** [comment]
Fixed in #739
